### PR TITLE
Fix issue where falsey settings in the config file are ignored

### PR DESCRIPTION
**Pull request by joshvfleming**
When the config file is read here:

https://github.com/rapportive/conker/blob/master/lib/conker.rb#L146

The settings are evaluated, so if you have something like this:

``` ruby
MY_COOL_SETTING: false
```

The `false` value becomes a `FalseClass` rather than a `String`. This causes the test for truthiness to believe that we haven't set the value at all.

Example scenario:

In constants.rb the constant is configured thus:

``` ruby
:MY_COOL_SETTING => optional(:type => :boolean, :default => true)
```

and the constants.yml file contans:

``` ruby
MY_COOL_SETTING: false
```

The resulting value of `MY_COOL_SETTING` is `true`, which is unexpected.

### CODE REVIEW

File: lib/conker.rb [hunk 1]
```diff
@@ -239,7 +239,11 @@ def interpret_value(value, type)
       type = type.to_sym if type
       case type
       when :boolean
-        value.to_s.downcase == "true" || value.to_i == 1
+        if [true, false].include? value
```
Comment (samstokes) [inline lib/conker.rb:242]: @joshvfleming I don't understand why this change was needed.  Doesn't it produce the same result as the previous code for `true` and `false`?

```
true.to_s.downcase == "true" # => true
false.to_s.downcase == "true" # => false
```

Comment (joshvfleming) [inline lib/conker.rb:242]: @samstokes It was falling through to

```
value.to_i == 1
```

which fails, because booleans don't respond to `to_i`.

Comment (samstokes) [inline lib/conker.rb:242]: Ha, good point!

**Comment by samstokes** [comment]
@joshvfleming sorry this is taking longer than expected.  The bug you've found stems from some wrong assumptions conker makes about the config source, introduced when it was converted from just supporting environment variables, to also supporting YAML and hashes.  I'm worried that simply fixing the bug, without correcting the wrong assumptions, would effectively encode those assumptions deeper into the logic, leading to a more confusing result down the road.  It's hard to design the proper fix in the snippets of time between fixing blocker bugs... :(

**Comment by joshvfleming** [comment]
@samstokes That's what I figured. I think it's okay to delay this actually. For me it was more an exercise to learn more about how it works. :)
